### PR TITLE
Adds CloudWatch Metric Based Autoscaling

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

https://github.com/AlexsLemonade/refinebio/issues/174

Adds autoscaling based on a new custom CW Metric based on the Nomad queue length. Also parameterizes some of the inputs. 

Needs a proper crunch test but I'm running the Neuroblastoma crunch already until tomorrow at least. I'm pretty sure this works though.

### CODE REVIEW

**Comment by Miserlou** [comment]
It works! Yay!

screen shot 2018-03-22 at 5 12 00 pm
screen shot 2018-03-22 at 5 12 07 pm
screen shot 2018-03-22 at 5 12 15 pm

File: README.md [hunk 1]
```diff
@@ -482,6 +482,12 @@ steps to configure (such as setting up Nomad job specifications and performing d
 ./deploy.sh
 ```
 
+### Autoscaling and Setting Spot Prices
+
+`refinebio` uses AWS Auto Scaling Groups to provide elastic capacity for large work loads. To do this, we use "Spot Requests". To find a good bid price for your instance type, use the [spot request page](https://console.aws.amazon.com/ec2sp/v1/spot/home?region=us-east-1) and then click on the **view pricing history** chart. Choose your instance type and then choose a big price that it slightly higher than the current price for your availability zone (AZ).
```
Comment (dongbohu) [inline README.md]: In `... a big price that it ...`, "it" probably should be "is". 
By the way, can you also cut this line into multiple lines? It seems so long.

File: README.md [hunk 2]
```diff
@@ -482,6 +482,12 @@ steps to configure (such as setting up Nomad job specifications and performing d
...
+
+Then set your `TF_VAR_client_instance_type`, `TF_VAR_spot_price` and `TF_VAR_max_clients` to configure your scaling instance types, cost and size. `TF_VAR_scale_up_threshold` and `TF_VAR_scale_down_threshold` define the queue lengths which trigger the scaling alarms, though you probably won't need to tweak these as much.
```
Comment (dongbohu) [inline README.md]: You may want to cut this line too.

File: infrastructure/instances.tf [hunk 3]
```diff
@@ -259,12 +267,80 @@ resource "aws_instance" "nomad_client_1" {
...
+    
+    name_prefix = "auto-client-"
+    image_id = "${data.aws_ami.ubuntu.id}"
+    instance_type = "${var.client_instance_type}"
+    security_groups = ["${aws_security_group.data_refinery_worker.id}"]
+    iam_instance_profile = "${aws_iam_instance_profile.data_refinery_instance_profile.name}"
+    depends_on = ["aws_internet_gateway.data_refinery", "aws_instance.nomad_server_1"]
+    user_data = "${data.template_file.nomad_client_script_smusher.rendered}"
+    key_name = "${aws_key_pair.data_refinery.key_name}"
+    spot_price    = "${var.spot_price}"
```
Comment (dongbohu) [inline infrastructure/instances.tf]: You may want to remove the extra space between "spot_price" and "=".

**Comment by Miserlou** [comment]
Addressed

**Pull request merged**